Phase2 Doc update (#38)
percona-platform/portal-doc: docs/checks.md

| Check Name | Description | Tier |
| :--------- | :---------- | :--- |
| **Check if Binaries are 32-bits** | Notifies if version_compatible_machine equals i686. | Registered, Paid |
| **MySQL Automatic User Expired Password** | Warns if MySQL automatic password expiry is not active. | Paid |
| **MySQL InnoDB flush method and File Format check** | Checks the following settings: **innodb_file_format**, **innodb_file_format_max**, **innodb_flush_method** and **innodb_data_file_path** | Paid |
| **MySQL Checks based on values of MySQL configuration variables** | Checks the following settings: **sync_binlog**, **log_bin, general_log**, **tmp_table_size**, **sql_mode**, **read_only**. | Registered,Paid |
| **MySQL Checks based on values of MySQL replication configuration variables** | Warns if replication is not configured correctly. | Registered,Paid |
| **MySQL Binary Logs checks, Local infile and SQL Mode checks** | Warns about non-optimal settings for Binary Log, Local Infile and SQL mode. | Registered, Paid |
| **MySQL Configuration Check** | Warns if parameters not following Percona best practices, for infile, replication threads and replica checksum. | Registered, Paid |

> | **MySQL InnoDB password lifetime** | Warns about password lifetime. | Registered, Paid |

Reviewer: ```suggestion
| **MySQL InnoDB password lifetime** | Warns if password expiration is not set and users' passwords have an infinite lifetime. | Registered, Paid |
```

| **MySQL InnoDB Strict Mode** | Warns if InnoDB strict mode is disabled, which could compromise data integrity. | Paid |
| **MySQL index sizes** | Warns if any have indexes larger than data.  This indicates sub-optimial schema and should be reviewed. | Paid |
| **MySQL replication configuration check** |  Checks if a replica is safely logging replicated transactions.| Paid |
| **MySQL Users With Granted Public Networks Access** | Notifies about MySQL accounts allowed to be connected from public networks. | Paid |
| **MySQL Secure Transport** | Warns if MySQL server allows unencrypted remote connections. | Registered, Paid |
| **MySQL User Check** | Runs a high-level check on user setup | Registered |
| **MySQL Advanced User Check** | Runs a detailed check on user setup | Paid |
| **MySQL Security Check for password policy** | Runs advisor checks on password policy. | Paid |
| **MySQL Security Check for remplication** | Runs advisor checks on replica account. | Paid |

> | **MySQL Replication privileges** | Warns if replication privileges is mixed with more elevated privileges. | Paid |

Reviewer: ```suggestion
| **MySQL Replication privileges** | Warns if replication privileges are mixed with more elevated privileges. | Paid |
```

| **MySQL tables without Primary Key** | Warns if  tables added without primary keys. | Paid |

> | **MySQL Test Database** | This check returns a notice if there is database with name ‘test’ or ‘test\_%’. | Registered, Paid |

Reviewer: ```suggestion
| **MySQL Test Database** | Notifies of any databases named **‘test’** or **‘test\_%’**. | Registered, Paid |
```

| **MySQL Timezone** | Checks if time zone is correctly loaded. | Registered, Paid |
| **MySQL Version** | Warns if MySQL, Percona Server for MySQL, or MariaDB version is not the latest one. | Anonymous, Registered, Paid |

### MongoDB checks

| Check Name | Description | Tier |
| :--------- | :---------- | :--- |
| **MongoDB Active vs Available Connections** | Checks the ratio between Active and Available connections. | Registered, Paid |
| **MongoDB Authentication** | Warns if MongoDB authentication is disabled. | Registered, Paid |
| **MongoDB Security AuthMech** | Warns if MongoDB is not using the default SHA-256 hashing as SCRAM authentication method. | Registered, Paid |
| **MonogDB IP Bindings** | Warns if MongoDB network binding is not set as recommended. | Paid |

> | **MonogDB CPU cores** | Warns if the number of CPU cores does not meet the minimum recommended requirements according to best practices | Registered, Paid |

Reviewer: ```suggestion
| **MonogDB CPU cores** | Warns if the number of CPU cores does not meet the minimum recommended requirements according to best practices. | Registered, Paid |
```

| **MongoDB maxSessions** | Warns if MongoDB is using more maxSessions value other than the default one | Registered, Paid |
| **MongoDB Read Tickets** | Warns if MongoDB is using more than 128 read tickets. | Registered, Paid |
| **MongoDB Replica Set Topology** | Warns if the Replica Set cluster has less than three members. | Paid |
| **MongoDB TaskExecutorPoolSize High** | Warns if MongoDB TaskExecutorPoolSize count is higher than available CPU cores. | Registered, Paid |
| **MongoDB Version** | Warns if MongoDB or Percona Server for MongoDB version is not the latest one. | Anonymous, Registered, Paid |
| **MongoDB Write Tickets** | Warns if MongoDB network is using more than 128 write tickets. | Registered, Paid |
| **MongoDB Configuration Write Tickets** | Warns if MongoDB is using more than 128 write tickets during runtime. | Registered, Paid |

### PostgreSQL checks

| Check Name | Description | Tier |
| :--------- | :---------- | :--- |
| **PostgreSQL Archiver is Failing** | Verifies if the archiver has failed. | Registered, Paid |
| **PostgreSQL Cache Hit Ratio** | Checks database hit ratio and complains when this is too low. | Registered, Paid |

> | **PostgreSQL Configuration Change Requires Restart/Reload** | Warns when a configuration was changed and requires a server restart/reload | Registered, Paid |

Author: ```suggestion
| **PostgreSQL Configuration Change Requires Restart/Reload** | Warns when a configuration was changed and requires a server restart/reload. | Registered, Paid |
```
